project-base/app/src/Component/Domain/DomainAwareSecurityHeadersSetter.php

```diff
@@ -29,7 +30,11 @@ public function onKernelResponse(ResponseEvent $event): void
             return;
         }
 
-        if (!$this->domain->isHttps()) {
+        try {
+            if (!$this->domain->isHttps()) {
+                return;
+            }
+        } catch (NoDomainSelectedException) {
```

Comment:
I'm not sure if this solution will just postpone the problem into the future.
It's really easy to implement a feature that will use the domain during maintenance page.
Isn't possible to resolve domain before the maintenance page subscriber?

Reply:
I had similar worries, we can brainstorm it @TomasLudvik